### ISSUE TITLE
[BUG] Process exited due to  TypeError: log is not iterable

### ISSUE THREAD

**Issue by andywojciechowski** [microsoft/playwright-dotnet#2295]
**Context:**
- Playwright Version: 1.25.0
- Operating System: Windows 10
- .NET version: .NET 6
- Browser: Chromium

**Code Snippet**

Help us help you! Put down a short code snippet that illustrates your bug and
that we can run and debug locally. For example:

```cs
using System.Threading.Tasks;
using Microsoft.Playwright;

class Program
{
    public static async Task Main()
    {
        using var playwright = await Playwright.CreateAsync();
        var sclocatorLocatorOptions = new SelectorsRegisterOptions()
            {
                Script = @"{
query(root, selector) {
    if (typeof isc !== 'undefined' && typeof isc.AutoTest !== 'undefined') {
        return isc.AutoTest.getElement(selector);
    }
    return null;
},
queryAll(root, selector) {
    if (typeof isc !== 'undefined' && typeof isc.AutoTest !== 'undefined') {
        var element = isc.AutoTest.getElement(selector);
        if (element !== null) {
            return [element];
        }
    }
    return [];
}
}"
            };
        await playwright.Selectors.RegisterAsync("customselector", sclocatorLocatorOptions);
        await using var browser = await playwright.Chromium.LaunchAsync();
        var page = await browser.NewPageAsync();
        // Base URL defined elsewhere
        await page.GotoAsync(baseUrl);
        // UsernameTextbox and UserName defined elsewhere
        await usernameTextbox.TypeAsync(userName);
    }
}
```

**Describe the bug**

I'm not sure if this is more of a dotnet bug or Node bug but I'm having issues using the latest version of playwright-dotnet. I'm trying to run a login test and on the call to type the username I'm running into the following error.

```
Microsoft.Playwright.PlaywrightException : Process exited
    C:\tests\bin\Debug\net6.0\.playwright\package\lib\server\dom.js:1025
      for (const entry of log) this._progress.logEntry(entry);
                          ^
TypeError: log is not iterable
    at InjectedScriptPollHandler._streamLogs (C:\tests\bin\Debug\net6.0\.playwright\package\lib\server\dom.js:1025:27)
    at runNextTicks (node:internal/process/task_queues:61:5)
    at processImmediate (node:internal/timers:437:9)
```

The only thing I'd say that is different about the usual setup is I'm using a custom selector since the JS library that the app that I'm testing uses does not make traditional locator methods easy and instead provides its own custom element selectors. I provided the script that I'm using for the custom selector in case that error could be related to the custom selector. I've encountered this error on multiple Playwright versions after 1.21.0 and I'm using that version as a workaround.

**Comment by mxschmitt** [comment]
I tried to run your reproducible but it was working for me. I even tho removed GotoAsync and TypeAsync. Is it always reproducible for you?

**Comment by andywojciechowski** [comment]
Yes this is always reproducible for me on 1.25.0. An interesting thing that I noticed is only tests that are using the custom locator are failing on 1.25.0. For reference the custom locator is for a JS library that is not widely used called SmartClient. Unfortunately I don't think I'll be able to give out the link to the website I'm trying to test without consulting my team but I'll also reach out to my team to see if anyone can help me create a sample SmartClient app to help with reproducing this.

I also mention that I'm running these on a work provided laptop so I'm not sure if there's the possibility that's something is being blocked on this laptop specifically that will need to be whitelisted?

**Comment by mxschmitt** [comment]
Most likely it depends on baseUrl and usernameTextbox, so if you are not able to provide us a repro which includes that, we unfortunately can't act on it.

**Comment by andywojciechowski** [comment]
I understand I'm unable to share the actual base URL that I'm using in my tests without approval from my team first. But I was able to also reproduce this using a barebones sample application that I should be able to share. I'll get that ready to share when I'm back at work on Monday then I'll update the sample code.

**Comment by mxschmitt** [comment]
If you don't want to share it publicly, you can also share it with me on our slack in a private chat. https://aka.ms/playwright-slack

**Comment by andywojciechowski** [comment]
@mxschmitt I updated the sample code with an example that's reproducible on my computer and I invited you to a private GitHub repo which contains the sample web app that I used for this Playwright example. The web app was originally written using .NET Core 3.1 but other than that you should be able to just build the web app solution and run it locally with no additional setup. Then while that solution is running you should be able to use the updated sample code I provided.

**Comment by mxschmitt** [comment]
Thanks for your repro, this was very helpful!

Turns out your site modifies the `globalThis.Window object` which makes Playwright missbehave. I'll take this as an opportunity to fix it on our side!

```
TypeError: Right-hand side of 'instanceof' is not callable
    at serialize (<anonymous>:99:38)
    at serializeAsCallArgument2 (<anonymous>:95:12)
    at UtilityScript.jsonValue (<anonymous>:206:12)
    at safeJson (<anonymous>:211:21)
    at UtilityScript._promiseAwareJsonValueNoThrow (<anonymous>:222:12)
    at UtilityScript.evaluate (<anonymous>:201:33)
    at UtilityScript.<anonymous> (<anonymous>:1:44)

globalThis.Window && value instanceof globalThis.Window
```
